ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerFragment.kt

```diff
@@ -130,28 +128,26 @@ class HomeScreenMapContainerFragment : AbstractMapContainerFragment() {
     super.onViewCreated(view, savedInstanceState)
     setupMenuFab()
     setupBottomLoiCards()
-    lifecycleScope.launch {
-      mapContainerViewModel.activeSurvey
-        .combine(mapContainerViewModel.lois) { x, y -> Pair(x, y) }
-        .collect {
-          val survey = it.first
-          val lois = it.second
-          survey?.let {
-            val suggest =
-              it.jobs.any {
-                it.strategy != Job.DataCollectionStrategy.PREDEFINED &&
-                  it.strategy != Job.DataCollectionStrategy.UNKNOWN
-              }
-            val readonly = lois.isEmpty()
-            val msg =
-              when {
-                suggest -> resources.getString(R.string.suggest_data_collection_hint)
-                readonly -> resources.getString(R.string.read_only_data_collection_hint)
-                else -> resources.getString(R.string.predefined_data_collection_hint)
-              }
-            Snackbar.make(binding.root, msg, Snackbar.LENGTH_LONG).show()
-          }
+    lifecycleScope.launch { showDataCollectionHint() }
+  }
+
+  /**
+   * Displays a popup hint informing users how to begin collecting data, based on the properties of
+   * the active survey.
+   *
+   * This method should only be called after view creation.
+   */
+  private suspend fun showDataCollectionHint() {
+    mapContainerViewModel.surveyUpdateFlow.collect {
```

Comment:
We should probably check `mapContainerViewModel != null` here and log a warning if this is called before view creation.

Reply:
sgtm, I was hoping android had an annotation for this (e.g. like their `WorkerThread` annotation), but alas.